Extension/src/main.ts

```diff
@@ -62,20 +62,15 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
     const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
     if (settings.intelliSenseEngine === "Disabled") {
         languageServiceDisabled = true;
-        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
-            if (!reloadMessageShown && settings.intelliSenseEngine !== "Disabled") {
-                reloadMessageShown = true;
-                util.promptForReloadWindowDueToSettingsChange();
-            }
-        }));
-    } else {
-        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
-            if (!reloadMessageShown && settings.intelliSenseEngine === "Disabled") {
-                reloadMessageShown = true;
-                util.promptForReloadWindowDueToSettingsChange();
-            }
-        }));
     }
+    const currentIntelliSenseEngineValue: string | undefined = settings.intelliSenseEngine;
+    disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+        if (!reloadMessageShown && settings.intelliSenseEngine !== currentIntelliSenseEngineValue) {
+            reloadMessageShown = true;
+            util.promptForReloadWindowDueToSettingsChange();
+        }
+    }));
     await LanguageServer.activate();
```

Comment:
Maybe in the future we pull any non-language server stuff out of `LanguageServer.activate` and just don't call it if the intelliSenseEngine is disabled.  I think that's a bigger refactor than what we want to do now (pulling task provider out of LanguageServer).

Reply:
This didn't seem too big, so I went ahead and did so.